core/sr-primitives/src/lib.rs

```diff
@@ -611,6 +616,62 @@ macro_rules! impl_outer_config {
 	}
 }
 
+#[macro_export]
+#[doc(hidden)]
+macro_rules! __parse_pattern_2 {
+	(PreRuntime $module:ident $internal:ident $v1:ident $v2:ident) => {
+		$internal::$module($module::RawLog::PreRuntime(ref $v1, ref $v2, $crate::PhantomData))
```

Comment:
one solution to avoid this special parse is to make a variant being a unique struct like this. (even if its boring to derive because of incorrect bound by standard derive macros)
```diff
diff --git a/srml/aura/src/lib.rs b/srml/aura/src/lib.rs
index 7cc7e2755..7cad009c2 100644
--- a/srml/aura/src/lib.rs
+++ b/srml/aura/src/lib.rs
@@ -102,13 +102,39 @@ pub type Log<T> = RawLog<T>;
 ///
 /// The type parameter distinguishes logs belonging to two different runtimes,
 /// which should not be mixed.
-#[cfg_attr(feature = "std", derive(Serialize, Debug))]
-#[derive(Encode, Decode, PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "std", derive(Serialize))]
+#[derive(Encode, Decode)]
+#[serde(bound(serialize = "T: Trait"))]
 pub enum RawLog<T> {
        /// AuRa inherent digests
-       PreRuntime([u8; 4], Vec<u8>, PhantomData<T>),
+       PreRuntime(PreRuntime<T>),
+}
+
+impl<T> std::fmt::Debug for RawLog<T> {
+       fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+               unimplemented!();
+       }
+}
+
+impl<T> Clone for RawLog<T> {
+       fn clone(&self) -> Self {
+               unimplemented!();
+       }
 }
 
+impl<T> Eq for RawLog<T> {
+}
+
+impl<T> PartialEq for RawLog<T> {
+       fn eq(&self, other: &Self) -> bool {
+               unimplemented!();
+       }
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Debug))]
+#[derive(Encode, Decode, PartialEq, Eq, Clone)]
+pub struct PreRuntime<T>(pub [u8; 4], pub Vec<u8>, pub PhantomData<T>);
+
 /// Provides the slot duration inherent data for `Aura`.
 #[cfg(feature = "std")]
 pub struct InherentDataProvider {
```

Reply:
OK, this change is legit as Log cannot be non-generic, maybe we could in the future introduce non-generic Log but this would require some change and should be made in another PR.